include/mapbox/variant_visitor.hpp

```diff
@@ -13,7 +15,7 @@ struct visitor<Fn> : Fn
     using type = Fn;
     using Fn::operator();
 
-    visitor(Fn fn) : Fn(fn) {}
+    visitor(Fn&& fn) : Fn(std::forward<Fn>(fn)) {}
```

Comment:
In this context, `Fn&&` is an rvalue reference, not a universal reference. I believe that this will cause constructs such as the following to stop working:

```
auto lambda = [](...) { ... };
rv.match(lambda);
```

Can you add a test for this?